waspc/src/Wasp/Generator/ServerGenerator/AuthG.hs

```diff
@@ -158,3 +154,38 @@ genProvidersTypes auth = return $ C.mkTmplFdWithData [relfile|src/auth/providers
     userEntityName = AS.refName $ AS.Auth.userEntity auth
 
     tmplData = object ["userEntityUpper" .= (userEntityName :: String)]
+
+genLuciaTs :: AS.Auth.Auth -> Generator FileDraft
+genLuciaTs auth = return $ C.mkTmplFdWithData [relfile|src/auth/lucia.ts|] (Just tmplData)
+  where
+    tmplData =
+      object
+        [ "sessionEntityLower" .= (Util.toLowerFirst DbAuth.sessionEntityName :: String),
+          "authEntityLower" .= (Util.toLowerFirst DbAuth.authEntityName :: String),
+          "userEntityUpper" .= (userEntityName :: String)
+        ]
+
+    userEntityName = AS.refName $ AS.Auth.userEntity auth
+
+genSessionTs :: AS.Auth.Auth -> Generator FileDraft
+genSessionTs auth = return $ C.mkTmplFdWithData [relfile|src/auth/session.ts|] (Just tmplData)
+  where
+    tmplData =
+      object
+        [ "userEntityUpper" .= userEntityName,
+          "userEntityLower" .= Util.toLowerFirst userEntityName,
+          "authFieldOnUserEntityName" .= DbAuth.authFieldOnUserEntityName,
+          "identitiesFieldOnAuthEntityName" .= DbAuth.identitiesFieldOnAuthEntityName
+        ]
+
+    userEntityName = AS.refName $ AS.Auth.userEntity auth
+
+depsRequiredByAuth :: AppSpec -> [AS.Dependency.Dependency]
+depsRequiredByAuth spec = maybe [] (const authDeps) maybeAuth
+  where
+    maybeAuth = AS.App.auth $ snd $ getApp spec
+    authDeps =
+      AS.Dependency.fromList
+        [ ("lucia", "^3.0.0-beta.14"),
```

Comment:
Lucia v3 is still now officially out, we are using the beta version here. The API shouldn't change with the release as per the author, so I think this should be fine to release with.

Reply:
Info from Lucia's author:

> What I can guarantee
> 1. No more API changes where you'd have to replace the overall code logic (you may need to replace e.g. a() with b())
> 2. No more changes to the database schema
>
> That said, I don't have any more planned changes to Lucia and Arctic

This puts in a good spot to carry on with this 👍